src/OrleansTestKit/Timers/TestTimer.cs

```diff
@@ -14,15 +14,26 @@ public TestTimer(Func<object, Task> asyncCallback, object state)
             _state = state;
         }
 
+        [Obsolete("Use FireAsync instead.")]
         public void Fire()
         {
             _asyncCallback?.Invoke(_state);
         }
 
+        public Task FireAsync()
+        {
+            if (_asyncCallback == null)
```

Comment:
It is not 100% equivalent of `?.` Should I add defense copy of delegate or there is no contention?

Reply:
I go back and forth on multi-threaded safety checks... There's an expectation that the grain will only be used on a single thread to match Orleans's behavior. However, it's really up to the author of the unit test to ensure it. It sounds nice that we'd guard against multi-threaded scenarios, but there are lots of places in the TestKit where things would break if a single fixture was used concurrently or in parallel.